src/rfq/protocols/bebop/client.rs

```diff
@@ -0,0 +1,429 @@
+        for pair in pairs {
+            pair_names.insert(pair_to_bebop_format(&pair));
+        }
+        Self { url, pairs: pair_names, chain, tvl, ws_user, ws_key }
+    }
+
+    /// Calculates Total Value Locked (TVL) based on bid/ask levels.
+    ///
+    /// TVL is calculated using the formula from Bebop's documentation:
+    /// https://docs.bebop.xyz/bebop/bebop-api-pmm-rfq/rfq-api-endpoints/pricing#interpreting-price-levels
+    ///
+    /// Returns the average of bid and ask TVLs across all price levels.
+    ///
+    /// Note: This calculation assumes all pairs use the same quote token.
+    /// For cross-pair comparisons, token prices should be normalized to a common denomination.
+    fn calculate_tvl(&self, price_data: &BebopPriceData) -> f64 {
+        let bid_tvl: f64 = price_data
+            .bids
+            .iter()
+            .map(|(price, size)| price * size)
+            .sum();
+
+        let ask_tvl: f64 = price_data
+            .asks
+            .iter()
+            .map(|(price, size)| price * size)
+            .sum();
+
+        (bid_tvl + ask_tvl) / 2.0
+    }
+
+    fn create_component_with_state(
+        &self,
+        component_id: String,
+        tokens: Vec<tycho_common::Bytes>,
+        price_data: &BebopPriceData,
+        tvl: f64,
+    ) -> ComponentWithState {
+        let protocol_component = ProtocolComponent {
+            id: component_id.clone(),
+            protocol_system: "rfq:bebop".to_string(),
+            protocol_type_name: "bebop_pool".to_string(),
+            chain: self.chain,
+            tokens,
+            contract_ids: vec![], // empty for RFQ
+            static_attributes: Default::default(),
+            change: Default::default(),
+            creation_tx: Default::default(),
+            created_at: Default::default(),
+        };
+
+        let mut attributes = HashMap::new();
+
+        // Store all bids and asks as JSON strings, since we cannot store arrays
+        if !price_data.bids.is_empty() {
+            let bids_json = serde_json::to_string(&price_data.bids).unwrap_or_default();
+            attributes.insert("bids".to_string(), bids_json.as_bytes().to_vec().into());
+        }
+        if !price_data.asks.is_empty() {
+            let asks_json = serde_json::to_string(&price_data.asks).unwrap_or_default();
+            attributes.insert("asks".to_string(), asks_json.as_bytes().to_vec().into());
+        }
+
+        ComponentWithState {
+            state: ResponseProtocolState {
+                component_id: component_id.clone(),
+                attributes,
+                balances: HashMap::new(),
+            },
+            component: protocol_component,
+            component_tvl: Some(tvl),
+            entrypoints: vec![],
+        }
+    }
+}
+
+#[async_trait]
+impl RFQClient for BebopClient {
+    fn stream(
+        &self,
+    ) -> BoxStream<'static, Result<(String, StateSyncMessage<TimestampHeader>), RFQError>> {
```

Comment:
A bit confused, if the StateSyncMessage contains ALL token pairs removed + changed, what is the String here? I guess not a component id? :/ @dianacarvalho1

Reply:
it's just "bebop"